File: easy/2176. Count Equal and Divisible Pairs in an Array/2176. Count Equal and Divisible Pairs in an Array.go
package main

func countPairs(nums []int, k int) int {
	res := 0
	n := len(nums)

	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			if nums[i] == nums[j] && (i*j)%k == 0 {
				res += 1
			}
		}
	}

	return res
}

File: easy/2176. Count Equal and Divisible Pairs in an Array/2176. Count Equal and Divisible Pairs in an Array.py
from typing import List


class Solution:
    def countPairs(self, nums: List[int], k: int) -> int:
        res = 0

        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] == nums[j] and (i * j) % 2 == 0:
                    res += 1

        return res


solution = Solution()
print(solution.countPairs(nums=[3, 1, 2, 2, 2, 1, 3], k=2))
print(solution.countPairs(nums=[1, 2, 3, 4], k=1))
